src/joystick/apple/SDL_mfijoystick.m

```diff
@@ -349,22 +349,24 @@ static BOOL IOS_AddMFIJoystickDevice(SDL_JoystickDeviceItem *device, GCControlle
             }
             if (controller.physicalInputProfile.buttons[GCInputXboxPaddleOne] != nil) {
                 device->has_xbox_paddles = SDL_TRUE;
-                device->button_mask |= (1 << SDL_GAMEPAD_BUTTON_PADDLE1);
+                device->button_mask |= (1 << SDL_GAMEPAD_BUTTON_RIGHT_PADDLE1);
                 ++nbuttons;
             }
             if (controller.physicalInputProfile.buttons[GCInputXboxPaddleTwo] != nil) {
+                /* TODO: Is this right? SDL_gamepad.h says P2 is the lower right */
                 device->has_xbox_paddles = SDL_TRUE;
-                device->button_mask |= (1 << SDL_GAMEPAD_BUTTON_PADDLE2);
+                device->button_mask |= (1 << SDL_GAMEPAD_BUTTON_LEFT_PADDLE1);
                 ++nbuttons;
             }
             if (controller.physicalInputProfile.buttons[GCInputXboxPaddleThree] != nil) {
+                /* TODO: Is this right? SDL_gamepad.h says P3 is the upper left */
                 device->has_xbox_paddles = SDL_TRUE;
-                device->button_mask |= (1 << SDL_GAMEPAD_BUTTON_PADDLE3);
+                device->button_mask |= (1 << SDL_GAMEPAD_BUTTON_RIGHT_PADDLE2);
                 ++nbuttons;
             }
```

Comment:
I don't have any Apple hardware, but I think the current code might be wrong here, swapping P2 and P3. For now I've preserved existing behaviour.

Reply:
The Elite controller is currently broken on iOS, (they no longer report paddles, and I've opened a bug report with Apple) but they were correct when I tested them in the initial implementation. I can re-verify once that bug is fixed.

Reply:
Huh. Maybe they think the order used by SDL is more correct, and swapped P2 and P3 in their implementation?